CONTRIBUTING.md

```diff
@@ -47,6 +47,61 @@ and additionally the following rules:
 * Use `my ($self) = @_;` for parameter parsing in methods when accessing the
   `$self` object. Do not parse any parameter if you do not need any.
 * [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)
+* Avoid `sleep()`: Do not use `sleep()` or simulate a sleep-like functionality.
+  This is introducing the risk of either unnecessarily wasting time or just
+  shifting race conditions and making them even harder to investigate. You *can*
+  use sleep but there are only very limited cases where is the best solution
+  (e.g. yield threads in low-level system code). Here you have the classical
+  problem in software design of
+  [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science))
+  . sleep should not be used in place of proper synchronization methods.
+  https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/
+  is one of multiple references that explains this: "The original problem is
+  likely a timing/synchronization issue, ignoring it by hiding it with
+  Thread.Sleep is only going to delay the problem and make it occur in random,
+  hard to reproduce ways." And if the problem does not occur at least time is
+  wasted.
+  We are waiting because some condition changes some time ... keyword(s) is/are
+  *some time*. It is important to know the exact condition that the code should
+  wait for and check for that instead of delaying program execution for an
+  arbitrary amount of time. For openQA tests in general just try to do what you
+  as a human user would also do. You would not look on your watch waiting for
+  exactly 100s before you look at the screen again, right? :)
+* Avoid `is_tumbleweed()`: Our tests should always consider openSUSE Tumbleweed
```

Comment:
Just a small comment here, as unfortunately Factory First workflow is not always followed, `is_tumbleweed` might appear attractive. Even though, decision made on version of the distribution doesn't scale well in general, as we get newer versions of the packages being available, as we could apply same mechanism of overrides, like we have for UI (see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/ui-framework-documentation.md).
So just in case you want to add this point.

Other than that LGTM, and I somehow missed this PR originally.

Reply:
Good point. I know your pain regarding this topic :)
I will add the pointer to ui-framework-documentation.md